tests/integration/utils.py

```diff
@@ -289,6 +289,56 @@ def test_endpoint(self, endpoint, resource, response_code, query_params=None,
     return response
 
 
+def test_filter_params(self, endpoint, filter_type, valid_tests,
+                       invalid_tests):
+    PARK_RES = 'ParkResource'
+    for test in valid_tests:
+        params = {f'filter[{filter_type}]': test}
+        response = test_endpoint(self, endpoint, PARK_RES, 200,
+                                 query_params=params)
+        response_data = response.json()['data']
+        for resource in response_data:
+            actual = resource['attributes']['location'][filter_type]
+            self.assertEqual(actual, test)
+    if filter_type != 'state':
+        for test in invalid_tests:
+            params = {f'filter[{filter_type}]': test}
+            response = test_endpoint(self, endpoint, PARK_RES, 200,
+                                     query_params=params)
+            response_data = response.json()['data']
+            self.assertFalse(response_data)
+    else:
+        for state in invalid_tests:
+            params = {f'filter[{filter_type}]': state}
+            test_endpoint(self, endpoint, 'ErrorObject', 400,
+                          query_params=params)
```

Comment:
Just a nitpick, but I think switch if and else will be easier to understand:

```python
if filter_type == 'state':
    for state in invalid_tests:
        do_something()
else:
    for test in invalid_tests:
        do_some_other_thing()
```

What do you think?

Reply:
I agree. It makes it more clear that the test is just for filter[state]